nifi-framework-bundle/nifi-framework/nifi-framework-components/src/main/java/org/apache/nifi/util/FlowDifferenceFilters.java

```diff
@@ -547,4 +577,302 @@ private static boolean isParameterContextChange(final FlowDifference flowDiffere
+        return (context == null || context.parameterizedPropertyRenames().isEmpty()) ? false : context.parameterizedPropertyRenames().contains(difference);
+    }
+
+    private static Optional<String> getComponentInstanceIdentifier(final FlowDifference difference) {
+        final Optional<String> identifierB = getComponentInstanceIdentifier(difference.getComponentB());
+        if (identifierB.isPresent()) {
+            return identifierB;
+        }
+
+        return getComponentInstanceIdentifier(difference.getComponentA());
+    }
+
+    private static Optional<String> getComponentInstanceIdentifier(final VersionedComponent component) {
+        if (component == null) {
+            return Optional.empty();
+        }
+
+        if (component instanceof InstantiatedVersionedComponent) {
+            final String instanceId = ((InstantiatedVersionedComponent) component).getInstanceIdentifier();
+            if (instanceId != null) {
+                return Optional.of(instanceId);
+            }
+        }
+
+        return Optional.ofNullable(component.getIdentifier());
+    }
+
+    private static Optional<String> getParameterReferenceValue(final FlowDifference difference, final boolean fromComponentA) {
+        final VersionedComponent primaryComponent = fromComponentA ? difference.getComponentA() : difference.getComponentB();
+        final VersionedComponent secondaryComponent = fromComponentA ? difference.getComponentB() : difference.getComponentA();
+
+        final Map<String, String> primaryProperties = getProperties(primaryComponent);
+        if (primaryProperties.isEmpty()) {
+            return Optional.empty();
+        }
+
+        final Map<String, String> secondaryProperties = getProperties(secondaryComponent);
+
+        final Optional<String> fieldNameOptional = difference.getFieldName();
+        if (fieldNameOptional.isPresent()) {
+            final String fieldName = fieldNameOptional.get();
+            final String propertyValue = primaryProperties.get(fieldName);
+            if (propertyValue != null) {
+                return Optional.of(propertyValue);
+            }
+        }
+
+        // Fallback: find a property unique to the primary component whose value is a parameter reference.
+        for (Map.Entry<String, String> entry : primaryProperties.entrySet()) {
+            final String propertyName = entry.getKey();
+            final String propertyValue = entry.getValue();
+            if (!isParameterReference(propertyValue)) {
+                continue;
+            }
+
+            if (!secondaryProperties.containsKey(propertyName)) {
+                return Optional.of(propertyValue);
+            }
+        }
+
+        return Optional.empty();
+    }
+
+    private static Map<String, String> getProperties(final VersionedComponent component) {
+        if (component == null) {
```

Comment:
Recommend refactoring this method to a single return an use of `if ... else if ...`